packages/core/tests/oob-mediation.test.ts

```diff
@@ -90,8 +91,11 @@ describe('out of band with mediation', () => {
     mediatorAliceConnection = await mediatorAgent.connections.returnWhenIsConnected(mediatorAliceConnection!.id)
     expect(mediatorAliceConnection.state).toBe(DidExchangeState.Completed)
 
-    // ========== Set meadiation between Alice and Mediator agents ==========
+    // ========== Set mediation between Alice and Mediator agents ==========
     const mediationRecord = await aliceAgent.mediationRecipient.requestAndAwaitGrant(aliceMediatorConnection)
+    const connectonRecord = await aliceAgent.connections.findById(mediationRecord.connectionId)
+    const tag = connectonRecord?.getTag('connectionType')
```

Comment:
Maybe you can use the newly created getConnectionType method to retrieve it.

Also would be nice to add a few tests in connection module to check these new methods for connection type setting.